tsl/src/continuous_aggs/create.c

```diff
@@ -853,35 +858,62 @@ caggtimebucket_validate(CAggTimebucketInfo *tbinfo, List *groupClause, List *tar
 					/* Origin is always 3rd arg for date variants. */
 					if (list_length(fe->args) == 3)
 					{
+						Node *arg = lthird(fe->args);
 						custom_origin = true;
+						/* this function also takes care of named arguments */
+						const_arg = check_time_bucket_argument(arg, "third");
 						tbinfo->origin = DatumGetTimestamp(
-							DirectFunctionCall1(date_timestamp,
-												castNode(Const, lthird(fe->args))->constvalue));
+							DirectFunctionCall1(date_timestamp, const_arg->constvalue));
 					}
 					break;
 				case TIMESTAMPOID:
 					/* Origin is always 3rd arg for timestamp variants. */
 					if (list_length(fe->args) == 3)
 					{
+						Node *arg = lthird(fe->args);
 						custom_origin = true;
-						tbinfo->origin =
-							DatumGetTimestamp(castNode(Const, lthird(fe->args))->constvalue);
+						const_arg = check_time_bucket_argument(arg, "third");
+						tbinfo->origin = DatumGetTimestamp(const_arg->constvalue);
 					}
 					break;
 				case TIMESTAMPTZOID:
 					/* Origin can be 3rd or 4th arg for timestamptz variants. */
 					if (list_length(fe->args) >= 3 && exprType(lthird(fe->args)) == TIMESTAMPTZOID)
 					{
 						custom_origin = true;
-						tbinfo->origin =
-							DatumGetTimestampTz(castNode(Const, lthird(fe->args))->constvalue);
+						if (IsA(lthird(fe->args), Const))
+						{
+							tbinfo->origin =
+								DatumGetTimestampTz(castNode(Const, lthird(fe->args))->constvalue);
+						}
+						/* could happen in a statement like time_bucket('1h', .., 'utc', origin =>
+						 * ...) */
+						else if (IsA(lfourth(fe->args), NamedArgExpr))
```

Comment:
I believe that the test cases added in cagg_ddl_common already cover this line, or at least that is what I see with lldb, so I am not sure why they are reported as not covered.

Reply:
oh no wait I'm talking about lines 910 downward.. 🤦‍♀️

Reply:
I guess these lines can't be covered by a test right now because the function `time_bucket(interval, timestamptz, origin => timestamptz)` is currently not allowed in cagg definitions. So I will remove these lines for now. We already have an issue for allowing all time_bucket variants in cagg definitions (https://github.com/timescale/timescaledb/issues/5453) so we can add this logic once we fix that one.